project-generation/helpers.go

```diff
@@ -8,46 +8,91 @@ import (
+	listOfArguments := make(ListOfArguments)
+	listOfArguments["appName"] = &Argument{
+		InputVal:  appName,
+		Context:   ctx,
+		Validator: ValidateAppName,
+	}
+	listOfArguments["projectType"] = &Argument{
+		InputVal:  projectType,
+		Context:   ctx,
+		Validator: ValidateProjectType,
+	}
+	listOfArguments["projectLocation"] = &Argument{
+		InputVal:  projectLocation,
+		Context:   ctx,
+		Validator: ValidateProjectLocation,
+	}
+	listOfArguments, err = ValidateArguments(listOfArguments)
 	if err != nil {
+		log.Event(ctx, "validation error", log.Error(err))
 		return "", "", "", "", err
 	}
-	offerPurge := !repositoryCreated
-	validatedProjectLocation, err = ValidateProjectLocation(ctx, unvalidatedProjectLocation, validatedAppName, offerPurge)
+	an = listOfArguments["appName"].OutputVal
+	pt = listOfArguments["projectType"].OutputVal
+	pl = listOfArguments["projectLocation"].OutputVal
+	listOfArguments = make(ListOfArguments)
+	goVerUnset := goVersion == "unset" || goVersion == ""
+	if goVerUnset && projectType != "generic-project" {
+		listOfArguments["goVersion"] = &Argument{
+			InputVal:  goVersion,
+			Context:   ctx,
+			Validator: ValidateGoVersion,
+		}
+		gv = listOfArguments["goVersion"].OutputVal
+	}
+	listOfArguments, err = ValidateArguments(listOfArguments)
 	if err != nil {
+		log.Event(ctx, "validation error", log.Error(err))
 		return "", "", "", "", err
 	}
 
-	if unvalidatedGoVersion == "unset" && validatedProjectType != "generic-project" {
-		validatedGoVersion, err = promptForInput(ctx, "Please specify the version of GO to use")
+	return an, pt, pl, gv, nil
+}
+
+func ValidateArguments(arguments map[string]*Argument) (validArguments map[string]*Argument, err error) {
+	fmt.Println("function ValidateArguments hit")
+	validArguments = make(ListOfArguments)
+	for key, value := range arguments {
+		validArguments[key] = value
+	}
+
+	for key, value := range arguments {
+		fmt.Println("key " + key)
+		fmt.Println("value " + value.InputVal)
+		validArguments[key].OutputVal, err = value.Validator(value.Context, value.InputVal)
 		if err != nil {
-			return "", "", "", "", err
+			log.Event(context.Background(), "validation error ", log.Error(err))
+			return nil, err
 		}
-	} else {
-		validatedGoVersion = unvalidatedGoVersion
 	}
 
-	return validatedAppName, validatedProjectType, validatedProjectLocation, validatedGoVersion, nil
+	return validArguments, err
 }
 
 // ValidateAppName will ensure that the app name has been provided and is acceptable, if not it will keep
 // prompting until it is
 func ValidateAppName(ctx context.Context, unvalidatedAppName string) (validatedAppName string, err error) {
-	if unvalidatedAppName == "unset" {
-		validatedAppName, err = promptForInput(ctx, "Please specify the name of the application, if this is a "+
+	if unvalidatedAppName == "unset" || unvalidatedAppName == "" {
```

Comment:
This function is not recursive, and so will not keep prompting as the comment suggests. As a result, the prompt for input response is also not being validated (at least, not explicitly by this function - the input might demand that its not an empty response, but i doubt its checking `!= "unset"`).

If this was recursive, and nothing would get past the If case without being valid, you could probably drop the else and either return `unvalidatedAppName` directly (though I appreciate this is less explicit) or at least inline the overriding of the validateAppName as the happy path

If a change is needed, consider whether it's needed in all usages of PromptForInput